### PR TITLE
Eliminate the predicate of the consumer of circular buffer in compute warp

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This PR unblocks `MmaOp` from being used in warp specialized kernel.

### CODE REVIEW

File: tests/cpp/test_matmul.cpp [hunk 1]
```diff
@@ -3787,8 +3787,8 @@ TEST_F(HopperMatmulTest, HSH_NT_128BSwizzle) {
 
   inlineMost();
 
-  tv0c->circularBuffer(stages, prefetch);
-  tv1c->circularBuffer(stages, prefetch);
+  tv0c->circularBuffer(stages, prefetch, WarpSpecialized(ParallelType::TIDy));
+  tv1c->circularBuffer(stages, prefetch, WarpSpecialized(ParallelType::TIDy));
```
Comment (zasdfgbnm) [inline tests/cpp/test_matmul.cpp:3791]: Don't know perf yet, I am unable to get a H200 machine this morning, will check later together with stmatrix+128B swizzle.

Comment (zasdfgbnm) [reply tests/cpp/test_matmul.cpp:3791]: Note that even though warp specialization is enabled on matmul, the kernel is still different from my manual prototyping kernel in https://github.com/NVIDIA/Fuser/pull/3354, so I would not be surprised if the perf of this branch is not as good as https://github.com/NVIDIA/Fuser/pull/3354. If this is the case, we will need a few more iterations to get the perf we want.

**Comment by zasdfgbnm** [comment]
!test

**Comment by zasdfgbnm** [comment]
!test

File: tests/cpp/test_matmul.cpp (hunk 1 above)
Comment (zasdfgbnm) [reply tests/cpp/test_matmul.cpp]: Get some perf number on H100:

Before (pipelined):
```C++
 Time (%)  Total Time (ns)  Instances  Avg (ns)  Med (ns)  Min (ns)  Max (ns)  StdDev (ns)                                                  Name

 --------  ---------------  ---------  --------  --------  --------  --------  -----------  ----------------------------------------------------------------------------------------------------
     36.9           155838          1  155838.0  155838.0    155838    155838          0.0  <unnamed>::nvfuser_none_f0_c0_r0_g0(<unnamed>::Tensor<<unnamed>::__half, (int)3, (int)3>, <unnamed>…
     21.5            90719          1   90719.0   90719.0     90719     90719          0.0  nvjet_hsh_128x256_64x4_2x1_v_bz_coopA_NTN
```

After (warp specialized):

```C++
 Time (%)  Total Time (ns)  Instances  Avg (ns)  Med (ns)  Min (ns)  Max (ns)  StdDev (ns)                                                  Name

 --------  ---------------  ---------  --------  --------  --------  --------  -----------  ----------------------------------------------------------------------------------------------------
     35.6           147710          1  147710.0  147710.0    147710    147710          0.0  <unnamed>::nvfuser_none_f0_c0_r0_g0(<unnamed>::Tensor<<unnamed>::__half, (int)3, (int)3>, <unnamed>…
     22.0            91584          1   91584.0   91584.0     91584     91584          0.0  nvjet_hsh_128x256_64x4_2x1_v_bz_coopA_NTN
```

Note that smem epilogue is not enabled in this test. I am just running it as is.

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**